cli/registry/client/fetcher.go

```diff
@@ -262,12 +262,18 @@ func (c *client) iterateEndpoints(ctx context.Context, namedRef reference.Named,
 }
 
 // allEndpoints returns a list of endpoints ordered by priority (v2, https, v1).
-func allEndpoints(namedRef reference.Named) ([]registry.APIEndpoint, error) {
+func allEndpoints(namedRef reference.Named, insecure bool) ([]registry.APIEndpoint, error) {
 	repoInfo, err := registry.ParseRepositoryInfo(namedRef)
 	if err != nil {
 		return nil, err
 	}
-	registryService, err := registry.NewService(registry.ServiceOptions{})
+
+	serviceOpts := registry.ServiceOptions{}
+	if insecure {
+		logrus.Debugf("allowing insecure registry for: %s", reference.Domain(namedRef))
+		serviceOpts.InsecureRegistries = []string{reference.Domain(namedRef)}
```

Comment:
The name was a bit confusing, so checked if `reference.Domain()` returned the port-number as well, but looks like it does 😅 👍  https://github.com/docker/distribution/blob/6170ac53daed40ebbebc9fe5f482495d01f09f12/reference/reference_test.go#L51-L55

Reply:
yes it does. i double-checked that b/c i had the same concern. :) thanks @thaJeztah!